modules/nf-core/samshee/main.nf

```diff
@@ -0,0 +1,48 @@
+process SAMSHEE {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'docker://community.wave.seqera.io/library/pip_samshee:733e11f3377fc2e3' :
+        'community.wave.seqera.io/library/pip_samshee:733e11f3377fc2e3' }"
+
+    input:
+    tuple val(meta), path(samplesheet)
+    val(validator_schema)              //optional
```

Comment:
It's tricky to fully support samshee's range of inputs here. 
Samshee can accept either 
 * a schema name e.g. `urn:samshee:illuminav2/v1`
 * a json string, e.g. `{"required": ["BCLConvert_Data"]}`
 * a URI pointing to a local file, e.g. `{"$ref": "file:./test.schema.json"}`
 * a URI pointing to a remote file, e.g.  `'{"$ref": "https://dataportal.lit.eu/schemas/litngscoresamplesheet/v0.1/litngscoresamplesheet.schema.json"}'`
 
plus it can accept multiple schemas. Especially when referencing a local file, value channels are not ideal, as when working with a cloud-based solution, the file needs to be provisioned by nextflow. And even when it's a remote URL, it might be preferable to have the download handled by nextflow. 

Maybe it could work to have two input channels, one accepting value, and one accepting files, such that a call to the samshee module looks like


```nextflow
SAMSHEE(
    [['id': 'samplesheet01'], path("samplesheet.csv")],
    ['{"required": ["Data"]}', '{"$ref": "urn:samshee:illuminav2/v1"}'],
    [path("file:./test.schema.json"), path(https://dataportal.lit.eu/schemas/litngscoresamplesheet/v0.1/litngscoresamplesheet.schema.json")]
)
```

WDYT? 

)

Reply:
Hi @grst I think this makes sense, I'll work on it